net/tcp/tcp_sendfile.c

```diff
@@ -320,41 +298,38 @@ static uint16_t sendfile_eventhandler(FAR struct net_driver_s *dev,
 
           tcp_setsequence(conn->sndseq, seqno);
 
+          /* Notify the device driver of the availability of TX data */
+
+          sendfile_txnotify(psock, conn);
+
           /* Update the amount of data sent (but not necessarily ACKed) */
 
           pstate->snd_sent += sndlen;
           ninfo("pid: %d SEND: acked=%" PRId32 " sent=%zd flen=%zu\n",
                 getpid(),
                 pstate->snd_acked, pstate->snd_sent, pstate->snd_flen);
         }
-      else
-        {
-          nwarn("WARNING: Window full, wait for ack\n");
-          goto wait;
-        }
     }
 
-  if (pstate->snd_sent >= pstate->snd_flen
```

Comment:
how about outstanding ack case?

Reply:
The check for outstanding acks is in line 186 in the changed code.

Reply:
I have just forced-pushed the changes concerning your remarks above. So the check for outstanding acks is moved from line 186 to 251 in the changed code.